pkg/dryrun/dryrun_utils.go

```diff
@@ -0,0 +1,264 @@
+			if !ok {
+				notArrayPrint(cmd, path, key)
+
+				isStatusMatch = false
+
+				continue
+			}
+
+			if len(typedV) == 0 {
+				continue
+			}
+
+			switch typedV[0].(type) {
+			case map[interface{}]interface{}, map[string]interface{}:
+				allMatch := true
+				// Check all input status are match
+				for i, obj := range typedV {
+					elementMatch := false
+
+					mapObj, ok := convertMapStringKey(obj).(map[string]interface{})
+					if !ok {
+						failedParsePrint(cmd, path)
+
+						continue
+					}
+
+					for _, actualObj := range actualArray {
+						mapActualObj, ok := convertMapStringKey(actualObj).(map[string]interface{})
+						if !ok {
+							failedParsePrint(cmd, path)
+
+							continue
+						}
+
+						// If an matched element is found in the actual status, break the current loop.
+						if compareStatusObj(cmd, true,
+							fmt.Sprintf("%s[%d]", path, i), mapObj,
+							mapActualObj) {
+							elementMatch = true
+
+							break
+						}
+					}
+
+					if !elementMatch {
+						allMatch = false
+
+						errorMatchAnyArrayPrint(cmd, fmt.Sprintf("%s[%d]", path, i))
+
+						break
+					}
+
+					successMatchArrayPrint(cmd, fmt.Sprintf("%s[%d]", path, i))
+				}
+
+				if !allMatch {
+					errorMatchArrayPrint(cmd, path)
+
+					isStatusMatch = false
+				} else {
+					successMatchArrayPrint(cmd, path)
+				}
+			default:
+				// Simple comparison of arrays eg []string
+				if reflect.DeepEqual(typedV, actualArray) {
+					successMatchPrint(cmd, path, typedV, actualArray)
+				} else {
+					errorMatchPrint(cmd, path, typedV, actualArray)
+
+					isStatusMatch = false
+				}
+			}
+
+		case map[string]interface{}:
+			if actualMap, ok := actualValue.(map[string]interface{}); ok {
+				// Recursive call for nested maps
+				isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, typedV, actualMap)
+			} else {
+				failedParsePrint(cmd, path)
+
+				isStatusMatch = false
+			}
+		case interface{}:
+			typeOfTypeV := reflect.TypeOf(typedV).String()
+			typeOfActual := reflect.TypeOf(actualValue).String()
+
+			if typeOfTypeV == "map[interface {}]interface {}" &&
+				typeOfActual == "map[interface {}]interface {}" {
+				mapObj, ok := convertMapStringKey(typedV).(map[string]interface{})
+				if !ok {
+					failedParsePrint(cmd, path)
+
+					continue
+				}
+
+				mapActualObj, ok := convertMapStringKey(actualValue).(map[string]interface{})
+				if !ok {
+					failedParsePrint(cmd, path)
+
+					continue
+				}
+
+				isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, mapObj, mapActualObj)
+
+				continue
+			}
+
+			if typeOfTypeV != typeOfActual {
+				cmd.Println(errorColor(fmt.Sprintf("The path '%s' does not match at all", path)))
+
+				continue
+			}
+
+			cmd.Println(errorColor(fmt.Sprintf("Unsupported type %T at path '%s' for key '%s'", typedV, path, key)))
+		default:
+			cmd.Println(errorColor(fmt.Sprintf("Unsupported type %T at path '%s' for key '%s'", typedV, path, key)))
```

Comment:
Is this ever reached in tests? I'd expect `case interface{}` to grab everything, and maybe we should just make it the `default` case.